File: typescript/examples/request-options.ts
import { RequestOptions, TwoFinanceClient, configFromEnv } from "../src/index";

async function main() {
  const client = new TwoFinanceClient(configFromEnv(process.env));
  const options: RequestOptions = {
    headers: { "X-Trace-ID": "trace-1" },
    idempotencyKey: "candles-upsert-001",
    query: { source: "sdk-example" },
    pagination: { page: 1, limit: 25 },
    timeoutMs: 5000,
    maxRetries: 1
  };

  const response = await client.analytics.post(
    "/analytics/candles:upsert",
    { symbol: "BTC-USDT" },
    options
  );
  console.log("response:", response);
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: python/examples/request_options.py
from twofinance_sdk_client import RequestOptions, TwoFinanceClient, config_from_env


def main() -> None:
    client = TwoFinanceClient(config_from_env())
    response = client.analytics.post(
        "/analytics/candles:upsert",
        {"symbol": "BTC-USDT"},
        RequestOptions(
            headers={"X-Trace-ID": "trace-1"},
            idempotency_key="candles-upsert-001",
            query={"source": "sdk-example"},
            page=1,
            limit=25,
            timeout=5.0,
            max_retries=1,
        ),
    )
    print("response:", response)


if __name__ == "__main__":
    main()
